api/v1alpha1/nodehealthcheck_types.go

```diff
@@ -122,15 +122,18 @@ type UnhealthyCondition struct {
 // NodeHealthCheckStatus defines the observed state of NodeHealthCheck
 type NodeHealthCheckStatus struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="observedNodes",xDescriptors="urn:alm:descriptor:com.tectonic.ui:observedNodes"
-	//ObservedNodes specified the number of nodes observed by using the NHC spec.selecor
-	ObservedNodes int `json:"observedNodes,omitempty"`
+	//ObservedNodes specified the number of nodes observed by using the NHC spec.selector
+	// +kubebuilder:default:=0
+	ObservedNodes int `json:"observedNodes"`
 
 	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="healthynodes",xDescriptors="urn:alm:descriptor:com.tectonic.ui:healthyNodes"
 	//HealthyNodes specified the number of healthy nodes observed
-	HealthyNodes int `json:"healthyNodes,omitempty"`
+	// +kubebuilder:default:=0
+	HealthyNodes int `json:"healthyNodes"`
 
 	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="inFlightRemediations",xDescriptors="urn:alm:descriptor:com.tectonic.ui:inFlightRemediations"
 	//InFlightRemediations records the timestamp when remediation triggered per node
+	//+optional
```

Comment:
wouldn't the `omitempty` make it optional in any case ? (i.e it will not exist in case it's empty)

Reply:
can't find it atm, but I read somewhere that it's best to do both